pkg/signature/kms/cliplugin/signer.go

```diff
@@ -213,6 +207,9 @@ type CryptoSigner struct {
 
 // CryptoSigner returns a wrapper around PluginClient.
 func (c PluginClient) CryptoSigner(ctx context.Context, errFunc func(error)) (crypto.Signer, crypto.SignerOpts, error) {
+	if err := ctx.Err(); err != nil {
+		return nil, nil, err
+	}
```

Comment:
it feels off to check this at the time of object creation instead of object usage.

Reply:
I'm thinking of this context as having a deadline that's specific for object initialization. If the caller wants to reuse the same context in method calls, then that's okay.

- https://github.com/sigstore/timestamp-authority/blob/3c5d80434f3235c52aca136de038406b24a95b83/cmd/fetch-tsa-certs/fetch_tsa_certs.go#L88

Reply:
For what reason would this have an error at this point?

Reply:
If the context already expired, then no reason to continue trying to initialize the CrytpoSigner.

Also this article talks about using contexts in initializer functions, and that reusing it for methods leads to confusion.

 - https://go.dev/blog/context-and-structs

Reply:
Also, does it make sense to call the `errFunc` here?